src/main/java/com/powsybl/openloadflow/network/util/ActivePowerDistribution.java

```diff
@@ -86,8 +86,8 @@ public Result run(LfGenerator referenceGenerator, Collection<LfBus> buses, doubl
             iteration++;
         }
 
-        final boolean movedBuses = initialP.entrySet().stream()
-                .anyMatch(e -> Math.abs(e.getKey().getTargetP() - e.getValue()) > P_RESIDUE_EPS);
+        final boolean movedBuses = Math.abs(initialP.entrySet().stream()
+                .mapToDouble(e -> e.getKey().getTargetP() - e.getValue()).sum()) > P_RESIDUE_EPS;
 
```

Comment:
If unlucky (mismatch very close to P_RESIDUE_EPS, and roundings you could be below the criteria)
Maybe > P_RESIDUE_EPS*0.9   to avoid rounding effects ?

Reply:
indeed, thanks